_posts/2022-01-03-quarkus-aws-cloudwatch_extension.adoc

```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+title: 'Creating a Quarkus extension for AWS CloudWatch'
+date: 2022-01-03
+tags: extension cloudwatch
+author: bennetelli
+---
+
+== Creating a Quarkus extension for AWS CloudWatch
+
+We recently had the situation that we wanted to log our Quarkus application logs to AWS CloudWatch. Basically it takes some time but is not a big deal. Adding a CloudWatch dependency, creating a Log Handler and push the logs to CloudWatch via the provided AWS CloudWatch API. But what if you want to share it with others? Of course you could put it on GitHub as part of your project so others can copy and paste it, but that’s not the most elegant way to share code with others.
+
+That’s why we implemented a Quarkus extension so others can use it more easily and don’t need to reinvent the wheel or need to copy paste things around.
+How do I do that? Here the Quarkiverse Hub comes in. Quarkiverse is a GitHub organization where developers can host and share their extension with others. There are several benefits you get for free when hosting the extension in Quarkiverse instead of using the good old way of doing all the things on your own. By using Quarkiverse you don’t need to build the artifact, release it with the Sonatype Nexus Manager (or something similar) and distribute it on Maven Central and other repositories.
+Quarkiverse comes with all these things so you can focus on implementing the extension itself. The following post describes what needs to be done to initialize, implement and share a CloudWatch Quarkus extension.
+
+If you want to use Quarkiverse to publish your extension on the Hub (which is what we recommend) and make use of all the advantages you get by using it, you simply need to open a new issue in the Quarkiverse GitHub organization. By doing this, most of the requirements are fulfilled already as there will be a template generated for you and you just have to implement your extension code. In case you are using an existing project as template, there are some requirements you need to take care of.
```

Comment:
Shouldn't it be an `Extension proposal` issue in the `quarkusio/quarkus` project instead of the Quarkiverse GitHub org?

Reply:
Thanks, you are right.